polkadot/node/subsystem-util/src/lib.rs

```diff
@@ -440,3 +442,210 @@ impl Validator {
 		Signed::sign(&keystore, payload, &self.signing_context, self.index, &self.key)
 	}
 }
+
+/// Object used for holding and computing assigned chunk indices for validators.
+pub struct ChunkIndexCacheRegistry(
+	LruMap<(BlockNumber, SessionIndex), (Vec<ChunkIndex>, Option<ClientFeatures>)>,
+);
+
+impl ChunkIndexCacheRegistry {
```

Comment:
Nit: Feels like this is different enough form the rest of this file you can put it in its own module:
subsystem-util/src/chunk_index_cache.rs